src/algorithms/util/matrixToListConverter.java

```diff
@@ -0,0 +1,35 @@
+package src.algorithms.util;
+import java.util.*;
+/**
+ * Supports method to convert an Adjacency Matrix into an Adjacency list
+ * Adjacency Matrix in the form of a 2D Integer Array int[][]
+ * Adjacency list in the form of a 2D Integer array int[][], where array[i] is a list of neighbour indexes
```

Comment:
Also, 1 more thing, the conventional implementation of adjacency list is to use linked list or a scaled-down version of one with just node pointers. While you don't have to code out both, it would be nice to just describe how one can code out this version if they wish instead.

here's a brief description that you can touch-up on:
We can create a private node class that serve as a wrapper of the original object and hold whatever data that comes alongside (eg weights), as well as additional node attribute(s) that serve as pointers to next (or prev) node.
Then, simply initialize an empty array of size v, each index representing an element, and the value is the head of the linked list (ie just the node).
Note: in the event where integers aren't used to denote elements or perhaps are sparse (eg 0, 1, 123456545), one can consider using a dictionary where the key is the element and value is the head node.